src/openforms/formio/formatters/custom.py

```diff
@@ -30,10 +32,23 @@ class AddressNLFormatter(FormatterBase):
 
     empty_values = ({},)
 
-    def format(self, component: Component, value: AddressValue) -> str:
+    def format(self, component: AddressNLComponent, value: AddressValue) -> str:
         value = value.copy()
         value.setdefault("houseLetter", "")
         value.setdefault("houseNumberAddition", "")
+
+        if self.as_html and (city := value.get("city")):
+            return mark_safe(
```

Comment:
Discussed on slack but `format_html` was not rendering the new lines. I could only achieve this by using `mark_safe` or by filtering the value in the template itself.

Reply:
`format_html(...).strip()` convers the safe string to a regular string, and then it gets escaped again, so that's a pitfall apparently :)